specfile/specfile.py

```diff
@@ -111,6 +149,43 @@ def __exit__(
     ) -> None:
         self.save()
 
+    def __deepcopy__(self, memodict: Dict[int, Any]):
+        """
+        Deepcopies the object, handling file-like attributes.
+        """
+        specfile = self.__class__.__new__(self.__class__)
+        memodict[id(self)] = specfile
+
+        for k, v in self.__dict__.items():
+            if k == "_file":
+                continue
+            setattr(specfile, k, copy.deepcopy(v, memodict))
+
+        try:
+            path = Path(self._file.name)
+        except AttributeError:
+            # not a named file, try `getvalue()`
+            try:
+                specfile._file = type(self._file)(self._file.getvalue())
```

Comment:
Here we are treating `self._file` as either `StringIO` or `BytesIO` (or their subclasses).

```suggestion
                specfile._file = type(cast(Union[StringIO, BytesIO], self._file))(
                    cast(Union[StringIO, BytesIO], self._file).getvalue()
                )
```

You could also define a new type for the union to make the code more compact.